src/DynamoCoreWpf/UI/GuidedTour/Step.cs

```diff
@@ -151,6 +152,18 @@ public void Hide()
             }
         }
 
+        /// <summary>
+        /// This method will update the Popup offset so it will be re-drawn in the new location (just when the PlacementTarget is moved or resized).
+        /// </summary>
+        public void UpdateLocation()
+        {
+            if(stepUIPopup.IsOpen == true)
+            {
+                stepUIPopup.HorizontalOffset = stepUIPopup.HorizontalOffset + 1;
+                stepUIPopup.HorizontalOffset = stepUIPopup.HorizontalOffset - 1;
```

Comment:
Is there a different way? Like mark it as modified ets? Using events?

Reply:
@QilongTang I changed the way the popup updates their locations, now I'm using reflection (the UpdatePosition() method in the Popup class is private), please let me know if this is ok since is the only way I found to do it.
https://github.com/DynamoDS/Dynamo/pull/11959/commits/c923ee41f84e324dd8d71b133e31219db5804ae7